go.mod

```diff
@@ -29,6 +29,7 @@ require (
 	github.com/spf13/cobra v1.8.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.9.0
+	go.akshayshah.org/memhttp v0.1.0
```

Comment:
We have a version of this in connect somewhere - we should use that instead, although I'm surprised we need something like this to test anyways

Reply:
If we are happy with the workaround jhump suggests in that issue I think this can be done as a unit test:
```go
type testRequest[T any] struct {
	*connect.Request[T]
}

func (r testRequest[_]) Spec() connect.Spec {
	return connect.Spec{
		Procedure: "/service/method",
		Addr: "...",
	}
}

// pass me to the interceptor as `next` arg.
func nextHandler(ctx context.Context, req conntext.AnyRequest) (conntext.AnyResponse, error) {
	return nil, connect.NewError(...)
}
```

Reply:
Replaced with the workaround in https://github.com/bufbuild/buf/pull/2871/commits/e6882be9d0abe78ef3e876aba181ec1eccb3ebbe, thanks for the suggestion!